drivers/display/display_gc9x01x.c

```diff
@@ -0,0 +1,701 @@
+		.len = 3U,
+		.data = {0x60U, 0x01U, 0x04U},
+	},
+	{
+		.cmd = 0xBEU,
+		.len = 1U,
+		.data = {0x11U},
+	},
+	{
+		.cmd = 0xE1U,
+		.len = 2U,
+		.data = {0x10U, 0x0EU},
+	},
+	{
+		.cmd = 0xDFU,
+		.len = 3U,
+		.data = {0x21U, 0x0CU, 0x02U},
+	},
+	{
+		.cmd = 0xEDU,
+		.len = 2U,
+		.data = {0x1BU, 0x0BU},
+	},
+	{
+		.cmd = 0xAEU,
+		.len = 1U,
+		.data = {0x77U},
+	},
+	{
+		.cmd = 0xCDU,
+		.len = 1U,
+		.data = {0x63U},
+	},
+	{
+		.cmd = 0x70U,
+		.len = 9U,
+		.data = {0x07U, 0x07U, 0x04U, 0x0EU, 0x0FU, 0x09U, 0x07U, 0x08U, 0x03U},
+	},
+	{
+		.cmd = 0x62U,
+		.len = 12U,
+		.data = {0x18U, 0x0DU, 0x71U, 0xEDU, 0x70U, 0x70U, 0x18U, 0x0FU, 0x71U, 0xEFU,
+			 0x70U, 0x70U},
+	},
+	{
+		.cmd = 0x63U,
+		.len = 12U,
+		.data = {0x18U, 0x11U, 0x71U, 0xF1U, 0x70U, 0x70U, 0x18U, 0x13U, 0x71U, 0xF3U,
+			 0x70U, 0x70U},
+	},
+	{
+		.cmd = 0x64U,
+		.len = 7U,
+		.data = {0x28U, 0x29U, 0xF1U, 0x01U, 0xF1U, 0x00U, 0x07U},
+	},
+	{
+		.cmd = 0x66U,
+		.len = 10U,
+		.data = {0x3CU, 0x00U, 0xCDU, 0x67U, 0x45U, 0x45U, 0x10U, 0x00U, 0x00U, 0x00U},
+	},
+	{
+		.cmd = 0x67U,
+		.len = 10U,
+		.data = {0x00U, 0x3CU, 0x00U, 0x00U, 0x00U, 0x01U, 0x54U, 0x10U, 0x32U, 0x98U},
+	},
+	{
+		.cmd = 0x74U,
+		.len = 7U,
+		.data = {0x10U, 0x85U, 0x80U, 0x00U, 0x00U, 0x4EU, 0x00U},
+	},
+	{
+		.cmd = 0x98U,
+		.len = 2U,
+		.data = {0x3EU, 0x07U},
+	},
+};
+
+static int gc9x01x_transmit(const struct device *dev, uint8_t cmd, const void *tx_data,
+			    size_t tx_len)
+{
+	const struct gc9x01x_config *config = dev->config;
+	int retval;
+	struct spi_buf tx_buf = {.buf = &cmd, .len = 1U};
+	struct spi_buf_set tx_bufs = {.buffers = &tx_buf, .count = 1U};
+
+	retval = gpio_pin_set_dt(&config->cmd_data, GC9X01X_GPIO_LEVEL_CMD);
+	if (retval < 0) {
+		return retval;
+	}
+	retval = spi_write_dt(&config->spi, &tx_bufs);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/* send data (if any) */
+	if (tx_data != NULL) {
+		tx_buf.buf = (void *)tx_data;
+		tx_buf.len = tx_len;
+
+		retval = gpio_pin_set_dt(&config->cmd_data, GC9X01X_GPIO_LEVEL_DATA);
+		if (retval < 0) {
+			return retval;
+		}
+		retval = spi_write_dt(&config->spi, &tx_bufs);
+		if (retval < 0) {
+			return retval;
+		}
+	}
+
+	return 0;
+}
+
+static int gc9x01x_regs_init(const struct device *dev)
+{
+	const struct gc9x01x_config *config = dev->config;
+	const struct gc9x01x_regs *regs = config->regs;
+	int retval;
+
+	/* Enable inter-command mode */
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_INREGEN1, NULL, 0);
+	if (retval < 0) {
+		return retval;
+	}
+	retval = gc9x01x_transmit(dev, GC9X01X_CMD_INREGEN2, NULL, 0);
+	if (retval < 0) {
+		return retval;
+	}
+
+	/* Apply default init sequence */
+	for (int i = 0; (i < ARRAY_SIZE(default_init_regs)) && (retval == 0); i++) {
+		retval = gc9x01x_transmit(dev, default_init_regs[i].cmd, default_init_regs[i].data,
+					  default_init_regs[i].len);
+		if (retval < 0) {
+			return retval;
+		}
+	}
+
+	/* Apply generic configuration */
+	retval =
```

Comment:
Maybe make this `ret` so that the indent is not needed?

Reply:
I didn't like that as well. Thanks for your porposal. Now the retail variable names are no longer consistent but I prefer readability as well.
FIxed in [5342707](https://github.com/zephyrproject-rtos/zephyr/commit/5342707641120c7b96c5ad63d62eb22b159f3b1b)

Reply:
> Now the retail variable names are no longer consistent

I think that we may be all suggesting to use `ret` in the whole file.

Reply:
OMG! It looks like I am meanwhile code blind and just react on comments.
Fixed in [c045f5c](https://github.com/zephyrproject-rtos/zephyr/commit/c045f5ce0a5f4aa4fb68a0bd7804e7944cc6564b)